src/java_bytecode/java_object_factory.cpp

```diff
@@ -494,6 +495,141 @@ class recursion_set_entryt
   }
 };
 
+/// Get max value for an integral type
+/// \param type:
+///   Type to find maximum value for
+/// \return Maximum integral valu
+static size_t max_value(const typet& type)
+{
+  if(type.id()==ID_signedbv)
+    return std::numeric_limits<int32_t>::max();
+  else if(type.id()==ID_unsignedbv)
+    return std::numeric_limits<uint32_t>::max();
+  UNREACHABLE;
+}
+
+/// Initialize a nondeterministic String structure
+/// \param obj: struct to initialize, must have been declared using
+///   code of the form:
+/// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+/// struct java.lang.String { struct @java.lang.Object;
+///   int length; char *data; } tmp_object_factory$1;
+/// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+/// \param loc: location in the source
+/// \param symbol_table: the symbol table
+/// \return code for initialization of the strings
+/// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+/// arg = &tmp_object_factory$1;
+/// tmp_object_factory$1.@java.lang.Object.@class_identifier =
+///   "java::java.lang.String";
+/// tmp_object_factory$1.@java.lang.Object.@lock = false;
+/// tmp_object_factory$1.length = NONDET(int);
+/// char tmp_object_factory$2[INFINITY];
+/// tmp_object_factory$2=NONDET(char[INFINITY])
+/// tmp_object_factory$1.data = tmp_object_factory$2;
+/// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+/// Unit tests in `unit/java_bytecode/java_object_factory/` ensure
+/// it is the case.
+codet initialize_nondet_string_struct(
+  const exprt &obj,
+  const std::size_t &max_nondet_string_length,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table)
+{
+  PRECONDITION(
+    java_string_library_preprocesst::
+      implements_java_char_sequence(obj.type()));
+
+  const namespacet ns(symbol_table);
+  code_blockt code;
+
+  // `obj` is `*expr`
+  // const dereference_exprt obj(expr, expr.type().subtype());
+  const irep_idt &class_id=
+    obj.type().id()==ID_symbol?
+    to_symbol_type(obj.type()).get_identifier():
+    irep_idt("java::"+id2string(to_struct_type(obj.type()).get_tag()));
```

Comment:
Taking reference to temporary!

Reply:
const reference, which should be fine (const reference prolongs the life of temporary only non-const ones are forbidden).